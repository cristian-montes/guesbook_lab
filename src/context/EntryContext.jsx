import { createContext, useContext } from "react";
import { useState } from "react/cjs/react.development";

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
    const [entries, setEntries] = useState([]);

    return <EntryContext.Provider value={{entries, setEntries}}> { children } </EntryContext.Provider>

}

const useEntries = () => {
    const context = useContext(EntryContext);

    if(context === undefined){
        throw Error('useEntries must be used within an EntryContext Provider');
    }

    return context
}

export { EntryProvider, useEntries}
