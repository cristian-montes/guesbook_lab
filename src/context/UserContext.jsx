import React from "react";
import { createContext, useContext } from "react";
import { useState } from "react/cjs/react.development";

const UserContext  = createContext();

const UserProvider = ({ children }) => {
    const [user, setUser] = useState('')
    return <UserContext.Provider value={{user, setUser}}> {children} </UserContext.Provider>
}

const useUser = () => {
    const context = useContext(UserContext);

    if(context === undefined){
        throw new Error('useUser hook must be called within a UserContext Provider')
    }

    return context;
}

export { UserProvider, useUser}