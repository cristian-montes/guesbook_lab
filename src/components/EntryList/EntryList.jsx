import React from "react";

import { useEntries } from "../../context/EntryContext";
import Entry from "../Entry/Entry";


const EntryList = () => {
    const { entries } = useEntries();
    return(
        <div>
            <ul>
                {entries.map((entry) => {
                    return(
                        <li key={`${entry.name}-${entry.message}`}>
                            <Entry entry={entry} />
                        </li>
                    )
                })}
            </ul>
        </div>
    )
}


export default EntryList;