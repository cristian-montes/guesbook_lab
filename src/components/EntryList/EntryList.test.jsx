import React from "react";
import { render, screen } from '@testing-library/react';
import EntryList from "./EntryList";
import { EntryProvider } from "../../context/EntryContext";

it('should display the list of entries', ()=> {
    const { container } = render(
    <EntryProvider>
        <EntryList />
    </EntryProvider> 
       )
    ;

    expect(container).toMatchSnapshot();
})
