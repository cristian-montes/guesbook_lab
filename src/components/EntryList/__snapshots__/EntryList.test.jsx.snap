// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the list of entries 1`] = `
<div>
   
  <div>
    <ul />
  </div>
   
</div>
`;
